### PR TITLE
Change string_val used in oc gnmi set to json_ietf_val

### PR DESCRIPTION
**Pull request by agvarghe-cisco**
Change string_val used in oc gnmi set to json_ietf_val
Encodings other than json_ietf_val are not supported

### CODE REVIEW

**Comment by liulk** [comment]
@hariramnat can you verify that this change will not break our existing tests?

By the way, I moved this test to a new location in #520.

File: feature/system/gnmi/cliorigin/mixed_oc_cli_origin_support_test/mixed_oc_cli_origin_support_test.go [hunk 1]
```diff
@@ -46,9 +46,10 @@ func buildOCUpdate(path *gpb.Path, value string) *gpb.Update {
 	if len(path.GetElem()) == 0 || path.GetElem()[0].GetName() != "meta" {
 		path.Origin = "openconfig"
 	}
+	json_value := fmt.Sprintf("\"%s\"", value)
 	update := &gpb.Update{
 		Path: path,
-		Val:  &gpb.TypedValue{Value: &gpb.TypedValue_StringVal{StringVal: value}},
+		Val:  &gpb.TypedValue{Value: &gpb.TypedValue_JsonIetfVal{JsonIetfVal: []byte(json_value)}},
```
Comment (hariramnat) [inline feature/system/gnmi/cliorigin/mixed_oc_cli_origin_support_test/mixed_oc_cli_origin_support_test.go:52]: This change might probably break our current tests. We usually run into this error: "error gathering JSON paths for coverage:json: cannot unmarshal string into Go value of type map[string]interface {}". Why cant we keep it as a StringVal itself? Why is this change necessary?

Comment (aaronmillisor) [reply feature/system/gnmi/cliorigin/mixed_oc_cli_origin_support_test/mixed_oc_cli_origin_support_test.go:52]: Neither 'json' nor 'string' have a canonical way to represent yang data. JSON_IETF is specified as RFC7951, and is the only format supported for yang data in gnmi for at least IOS XR. If 'cli' data is being set, 'string' can be used.

File: feature/system/gnmi/cliorigin/mixed_oc_cli_origin_support_test/mixed_oc_cli_origin_support_test.go [hunk 2]
```diff
@@ -46,9 +46,10 @@ func buildOCUpdate(path *gpb.Path, value string) *gpb.Update {
 	if len(path.GetElem()) == 0 || path.GetElem()[0].GetName() != "meta" {
 		path.Origin = "openconfig"
 	}
+	json_value := fmt.Sprintf("\"%s\"", value)
```
Comment (Ankur19) [inline feature/system/gnmi/cliorigin/mixed_oc_cli_origin_support_test/mixed_oc_cli_origin_support_test.go:49]: Our tests use camel case, can you change this to something like `jsonVal`?

Comment (Ankur19) [reply feature/system/gnmi/cliorigin/mixed_oc_cli_origin_support_test/mixed_oc_cli_origin_support_test.go:49]: The ygot way to convert to Json_ietf would be

`jsonVal, err := ygot.Marshal7951(ygot.String(value))`

**Review by Ankur19 — CHANGES_REQUESTED** [review]
Just minute nits and then we can merge the changes.

**Comment by agvarghe-cisco** [comment]
Raised #703

**Comment by Ankur19** [comment]
Also resolve conflicts by rebasing to main.

**Review by Ankur19 — APPROVED** [review]

**Comment by Ankur19** [comment]
Closed in favour of #703
